db/hybrid_logical_vector.go

```diff
@@ -180,6 +183,24 @@ func (hlv *HybridLogicalVector) GetVersion(sourceID string) uint64 {
 	return latestVersion
 }
 
+// AddNewerVersions will take a hlv and add any newer source/version pairs found across CV and PV found in the other HLV taken as parameter
+func (hlv *HybridLogicalVector) AddNewerVersions(otherVector HybridLogicalVector) {
+	if otherVector.PreviousVersions != nil || len(otherVector.PreviousVersions) != 0 {
+		_, localVersion := hlv.GetCurrentVersion()
+		// Iterate through incoming vector previous versions, compare to current version cas of local HLV
+		// Any more recent versions in incoming vectors previous versions need to be added to local HLV previous versions
+		for i, v := range otherVector.PreviousVersions {
+			if v > localVersion {
```

Comment:
We want to update with the `version` from otherVector for `source` if hlv's version for that `source` is lower.  There isn't a requirement that this value is greater than currentVersion.